### DISCUSSION TITLE
[2024-04-02] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#9597] [Weekly Demo Day]
- James: Snowflake Connectivity #9435
- James: Some fixes working towards a sales demo #9501
- James: Fix replace widget. #9505
- James: Data link for Snowflake. #9514
- James: Apply ICONs and other tweaks. #9552
- James: Restructure Standard.Table. #9559
- James: Avoiding trying to add a specific import if all already present. #9568
- Radosław: Make data-links behave more like 'symlinks' #9485
	- Follow-up for the data-link-to-normal-file copy scenario for later: #9522
- Radosław: Data Link for reading `Enso_File` #9525
- Radosław: Fix missing AWS region in S3 operations #9546
	- Ensures `AWS_Credential.Key` and DataLinks actually work on a clean environment without needing env vars setup, ensures that our CI tests such a scenario to avoid future regressions.
- Radosław: Use `DELETE /assets/<id>` for deleting files and secrets #9551
	- Using the endpoint recommended from the cloud team (before we used `DELETE /files/<id>` and `/secrets/<id>` instead)
- Radosław: Refactoring `Enso_File` to be path based #9581
	- Also updated our endpoints to changes in Cloud API (names of user and org ID related fields)
	- Support for `ENSO_PROJECT_PATH` that is setting the current working directory in the cloud.
- Radosław: Finalized the first iteration of the Type Inference PoC - it is now in review
  #8652 
    - Once greenlit, it will still have to wait for #9361 to be able to merge it, but I will still be able to start work on stage 2 based on this pending PR.
    - Challenges to be tackled in Stage 2 (and 3...; in no particular order):
      - resolving methods on known Types (currently we only resolve Atom Constructors, but not methods) - this will expand the amount of types that we infer the most.
      - handling of default arguments (currently if a constructor or method has default arguments we give up),
      - resolving static methods in module scope (currently only methods in local scope are resolved)
      - handling recursion, circular dependencies between modules (which will need to introduce some changes to the very simple bottom-up type reconstruction algorithm, to allow for some form of unification for solving simple type constraints)
      - statically checking for availability of `.from` conversions - this is needed to be able to raise type warning (a warning can only be raised if the types mismatch **and** there's **no** conversion that could reconcile the types).
- Greg: Conversions from `Decimal` to `Integer` and `Float`. #9462
- Greg: `Set.to_text` should not assume contents are `Text` #9548
- Greg: Decimal remainder, pow, div #9566
- Greg: `Decimal` arithmetic benchmarks #9582
- Greg: Add range info to `Out_Of_Range` error #9586
- Greg: Remove pending tests for hash consistency. #9588
- Greg: Make the`%` consistent across both `Integer` sizes. #9589
- Cassandra: Updated fraud blog
- Cassandra: Finished Google Analytics blog #8537 
- Cassandra: Finished data cleansing and remove duplicates video
- Cassandra: Added icons #8928 
- Ad: Add new aliases for take #9500
- Ad: Make expand_to_rows, expand_column support Rows, Tables, Column data types #9533
- Ad: Added icon for rows #9535
- Ad: Add google analytics sample data #9571
- Ad: Wip/adr/add more table running functionality #9577
- Jaroslav: [Towards Incremental Compilation](https://docs.google.com/document/d/1bKX2swSW-6ACLkTt623adrT45-SeANom4vv3w6XvqKc) strategic vision
- Jaroslav: Make sure import and FQN behave the same #9539
- Jaroslav: Don't let the non-Enso types float around the Enso interpreter! #9584
- Jaroslav: Removing useless methods #9590
- Jaroslav: Instrumentation, visualization and (autoscoped) constructors #9452
- Hubert: Make sure `std-benchmarks` has all dependencies #9504
- Hubert: Fix inline compilation benchmarks and Improve searching Graph.Link performance #9520
- Hubert: Potentially fix further `std-benchmarks` issues #9541
- Hubert: Add missing API for Jackson_Object #9591
- Hubert: Fix expectations in lock manager test #9592
- Hubert: Snowflake with Arrow investigation #9475 
- Pavel: Investigate Oracle GraalVM translated exception issue #9481
- Pavel: Add IRDumper for Scala IR #9543
- Pavel: Subprocess does not freeze when it generates a lot of output to stdout #9550
- Pavel: Add --disable-private-check runner option #9556
- Pavel: Equality does not swallow errors #9560
- Dmitry: Put all reexports of given entry in Suggestion Database #9454
- Dmitry: Fix getters serialization #9480
- Dmitry: Allow project manager to write files #9483
- Dmitry: Project manager create method should handle relative paths #9484
- Dmitry: Project manager returns normalized name of the created project #9510
- Dmitry: Run ydoc-server with GraalVM #9528

**Comment by radeusgd** [comment]
"Demo" of what is the 'visible' part in the Type Inference PoC 1:
![image](https://github.com/enso-org/enso/assets/1436948/edeb6aa4-4160-47bf-8878-3ab71334fed8)
